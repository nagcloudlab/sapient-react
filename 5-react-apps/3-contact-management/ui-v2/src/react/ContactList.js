import React from 'react';
import {Link, Route} from "react-router-dom";
import ContactView from "./ContactView";

import {useSelector, useDispatch} from 'react-redux'

function ContactList({}) {
    const contacts = useSelector(state => state.contacts)
    const dispatch = useDispatch();

    const handleDelete = id => {
        let action = {type: 'DELETE_CONTACT', id}
        dispatch(action)
    }

    const renderContacts = () => {
        return contacts.map(contact => {
            return (
                <tr key={contact.id}>
                    <td><Link to={`/all/view/${contact.id}`}>{contact.name}</Link></td>
                    <td><Link to={`/edit/${contact.id}`}><i className={"fa fa-edit"}></i></Link></td>
                    <td><Link to={""} onClick={e => {
                        e.preventDefault();
                        handleDelete(contact.id)
                    }}><i className={"fa fa-trash"}></i></Link></td>
                </tr>
            )
        })
    }
    return (
        <div>
            <table className={"table"}>
                <tbody>
                <tr>
                    <th>Name</th>
                    <th></th>
                    <th></th>
                </tr>
                {renderContacts()}
                </tbody>
            </table>
            <hr/>
            <Route path={"/all/view/:contactId"} render={props => <ContactView {...props} />}/>
        </div>
    );
}

export default ContactList;