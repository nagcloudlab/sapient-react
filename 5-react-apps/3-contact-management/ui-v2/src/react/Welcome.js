import React from 'react';

function Welcome(props) {
    return (
        <div>welcome to CM</div>
    );
}

export default Welcome;