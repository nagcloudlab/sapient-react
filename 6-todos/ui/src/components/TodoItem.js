import React, {useState} from 'react';

function TodoItem({value: todo, onDelete, onComplete, onEdit}) {
    let [isEditing, setEditing] = useState(false)
    let [title, setTitle] = useState(todo.title)
    let classNames = `list-group-item ${todo.completed ? 'bg-success' : ''}`
    const handleKeyUp = e => {
        if (e.key === "Enter") {
            setEditing(false)
            onEdit(todo.id,title)
        }
    }
    return (
        <li className={classNames}>
            <div className="d-flex justify-content-between">
                <input type="checkbox" onChange={e => onComplete(todo.id)} checked={todo.completed}/>
                {!isEditing ? <span style={{width: '100%', textAlign: 'center'}}
                                    onDoubleClick={e => setEditing(true)}>{todo.title}</span> : null}
                {isEditing ? <input value={title} onChange={e=>setTitle(e.target.value)} onKeyUp={e => handleKeyUp(e)}/> : null}
                <button onClick={e => onDelete(todo.id)} className="btn btn-sm btn-danger">delete</button>
            </div>
        </li>
    );
}

export default TodoItem;