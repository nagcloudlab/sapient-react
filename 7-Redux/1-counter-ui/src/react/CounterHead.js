import React from 'react';

import {useSelector} from 'react-redux'

function CounterHead({title}) {
    const count=useSelector(state=>state.counter.count)
    return (
        <div className={"alert alert-dark"}>
            <div data-testid={"title"}>
               <div className={"display-3"}>
                   Counter - <span className={"badge bg-danger"}>{count}</span>
               </div>
            </div>
        </div>
    );
}

export default CounterHead;