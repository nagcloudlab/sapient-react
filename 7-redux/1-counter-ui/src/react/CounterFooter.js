import React from 'react';


import {useSelector,useDispatch} from 'react-redux'

function CounterFooter({}) {
    const count=useSelector(state=>state.counter.count)
    const dispatch=useDispatch()
    const handleReset=e=>{
        let action={type:'RESET'}
        dispatch(action)
    }
    return (
        <div className={"alert alert-danger"}>
            <div data-testid={"count"}>
                <button onClick={e=>handleReset()} className={"btn btn-warning"}>reset</button>
                &nbsp;
                {count}
            </div>
        </div>
    );
}

export default CounterFooter;