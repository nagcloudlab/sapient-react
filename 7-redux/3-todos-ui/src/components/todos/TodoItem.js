import React from 'react';

function TodoItem({value:todo}) {
    return (
        <>
            <tr>
                <td>{todo.id}</td>
                <td>{todo.title}</td>
                <td>{todo.completed ? 'completed' : 'not completed'}</td>
                <td>{todo.userId}</td>
            </tr>
        </>
    );
}

export default TodoItem;