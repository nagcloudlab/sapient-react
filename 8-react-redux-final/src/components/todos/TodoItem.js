import React from 'react';

function TodoItem({value: todo, onDelete}) {
    return (
        <tr>
            <td data-testid="todoid">{todo.id}</td>
            <td>{todo.title}</td>
            <td>{todo.completed ? 'completed' : 'not completed'}</td>
            <td>{todo.userId}</td>
            <td>
                <button
                    data-testid={"delbtn"}
                    onClick={e => onDelete(todo.id)}
                    className={"btn btn-danger"}>
                    delete
                </button>
            </td>
        </tr>
    );
}

export default TodoItem;


