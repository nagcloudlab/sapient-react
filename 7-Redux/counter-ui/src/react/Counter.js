import React, {useState} from 'react';
import CountDisplay from "./CounterFooter";
import CounterHead from "./CounterHead";

import {useDispatch} from "react-redux";

function Counter(props) {
    // const [count,setCount]=useState(0)
    const [stepCount,setStepCount]=useState(1)
    const dispatch=useDispatch();
    const handleIncEvent=e=>{
        dispatch({type:'INCREMENT',value:stepCount})
    }
    const handleDecEvent=e=>{
        dispatch({type:'DECREMENT',value:stepCount})
    }
    return (
        <div className={"card card-body"} data-testid={"root"}>
            <CounterHead title={"count"} />
            <br/>
            <div className={"d-flex"}>
                <button className={"btn btn-lg btn-danger"}
                        data-testid={"sub-btn"}
                        onClick={handleDecEvent}
                >
                    -
                </button>
                <input
                    data-testid={"input"}
                    value={stepCount}
                    type={"number"}
                    onChange={e=>setStepCount(Number.parseInt(e.target.value))}
                />
                <button className={"btn btn-lg btn-success"}
                        data-testid={"add-btn"}
                        onClick={handleIncEvent}
                >
                    +
                </button>
            </div>
            <br/>
            <CountDisplay/>
        </div>
    );
}

export default Counter;