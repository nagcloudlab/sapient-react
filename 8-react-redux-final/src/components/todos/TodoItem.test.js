// import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// import { shallow } from 'enzyme';
// import TodoItem from "./TodoItem";
// Enzyme.configure({ adapter: new Adapter() });

import {render, fireEvent, screen} from '@testing-library/react'
import TodoItem from "./TodoItem";
import renderer from 'react-test-renderer';
/*

    TodoItem
    -----------
    props

        - value ( todo-data )
        - onDelete

     expectations

        - should render properly
        - should render given value prop
        - should render 'completed' if todo already completed
        - should render 'not completed' if todo not completed
        - should call 'onDelete' when delete btn clicked with todo id

 */

/*

what we need ?

    => Testing Framework:

        - Jest

    => additional libraries to test react-components with Jest

        - Enzyme
        -or-
        - React Testing Lib ( New )

 */


describe("TodoItem", () => {

    // way-1 : Enzyme

    // it("should render properly",()=>{
    //     let todo={id:1,title:'sample',completed:false,userId:1}
    //     const wrapper = shallow(<TodoItem value={todo}/>);
    //     const ele=wrapper.find('tr')
    //     expect(ele).toBeTruthy()
    // })

    // it("should render given prop",()=>{
    //     let todo={id:1,title:'sample',completed:false,userId:1}
    //     const wrapper = shallow(<TodoItem value={todo}/>);
    //     const idEle=wrapper.find('.todo-id')
    //     expect(Number.parseInt(idEle.text())).toBe(todo.id)
    // })

    // way-2 : react testing lib

    it("should render given prop", () => {
        let todo = {id: 1, title: 'sample', completed: false, userId: 1}
        render(<TodoItem value={todo}/>)
        const ele=screen.getByTestId("todoid")
        expect(ele).toHaveTextContent("1")
    })

    it("should call 'onDelete' when delete btn clicked with todo id",()=>{
        const mockCallback = jest.fn(x => {});
        let todo = {id: 123, title: 'sample', completed: false, userId: 1}
        render(<TodoItem value={todo} onDelete={mockCallback}/>)
        const delBtn=screen.getByTestId("delbtn")
        fireEvent.click(delBtn)
        expect(mockCallback.mock.calls.length).toBe(1)
        expect(mockCallback.mock.calls[0][0]).toBe(todo.id)
    })


    // snapshot testing

    it('renders correctly', () => {
        const mockCallback = jest.fn(x => {});
        let todo = {id: 123, title: 'sample', completed: false, userId: 1}
        const tree = renderer
            .create(<TodoItem value={todo} onDelete={mockCallback}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });



})
