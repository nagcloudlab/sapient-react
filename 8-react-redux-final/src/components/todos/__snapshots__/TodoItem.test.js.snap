// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem renders correctly 1`] = `
<tr>
  <td
    data-testid="todoid"
  >
    123
  </td>
  <td>
    sample
  </td>
  <td>
    not completed
  </td>
  <td>
    1
  </td>
  <td>
    <button
      className="btn btn-danger"
      data-testid="delbtn"
      onClick={[Function]}
    >
      delete
    </button>
  </td>
</tr>
`;
